File: 2021/day01.rb
vals = File.read(File.basename(__FILE__, ".rb") + ".input")

vals = vals.chomp.split.map(&:to_i)

p vals.each_cons(2).count { |x0, x1| x1 > x0 }
p vals.each_cons(3).map(&:sum).each_cons(2).count { |a, b| b > a }

File: 2022/day01.py
from inspect import cleandoc
from itertools import combinations

import os
import math

test_data = cleandoc("""
1000
2000
3000

4000

5000
6000

7000
8000
9000

10000
""")

base = os.path.splitext(os.path.basename(__file__))[0]
test_data = open(base + ".input").read()

cals = sorted([sum(map(int, x.split())) for x in test_data.split("\n\n")])
print(cals[-1])
print(sum(cals[-3:]))

